implementations/rust/ockam/ockam_api/src/nodes/service.rs

```diff
@@ -491,9 +503,9 @@ impl NodeManager {
 
         debug!("connected to {connection_instance:?}");
 
-        // Every piece of the chain must be part of the session to allow communication
-        for service in intermediary_services {
-            connection_instance.add_consumer(&service);
+        // make sure the service before flow control is a consumer
+        if let Some(address) = consumer_worker {
```

Comment:
Are there any use cases for that? I had an impression that we can just get rid of it

Reply:
In practice, I think It wouldn't break anything (other than `hop`), but I reckon that if the user provides an extra address in the `MultiAddr` we should do anything possible to make sure it's supported out of the box.